lib/SILOptimizer/Mandatory/DiagnoseStaticExclusivity.cpp

```diff
@@ -313,6 +313,26 @@ static StringRef extractExprText(const Expr *E, SourceManager &SM) {
   return SM.extractText(CSR);
 }
 
+/// Returns true when the call expression is a call to swap() in the Standard
+/// Library.
+/// This is a helper function that is only used in an assertion, which is why
+/// it is in a namespace rather than 'static'.
```

Comment:
Nitpick: I think better style would just be to check NDEBUG. Restructuring code to avoid the warnings Clang happens to have implemented today doesn't seem like a good long-term strategy.